File: Python3/Medium/Accepted/people-whose-list-of-favorite-companies-is-not-a-subset-of-another-list_2.py
class Solution:
    def peopleIndexes(self, favoriteCompanies: List[List[str]]) -> List[int]:
        
        favoriteCompanies = [set(x) for x in favoriteCompanies]
        res = []
        for i, x in enumerate(favoriteCompanies):
            if all(not x.issubset(y) for j, y in enumerate(favoriteCompanies) if i != j):
                res.append(i)
        return res

File: Java/Medium/TimeLimitExceeded/people-whose-list-of-favorite-companies-is-not-a-subset-of-another-list_2.java
class Solution {
    public List<Integer> peopleIndexes(List<List<String>> favoriteCompanies) {
        
        List<Integer> result = new ArrayList<>();
        for(int i = 0; i < favoriteCompanies.size(); i++){
            boolean isSubset = false;
            for(int j = 0; j < favoriteCompanies.size(); j++){
                if(i != j){
                    if(favoriteCompanies.get(i).size() < favoriteCompanies.get(j).size()){
                        if(favoriteCompanies.get(j).containsAll(favoriteCompanies.get(i))){
                            isSubset = true;
                            break;
                        }
                    }
                }
            }
            if(!isSubset){
                result.add(i);
            }
        }
        return result;
    }
}